### PR TITLE
Assorted T3 fixes

### PR DESCRIPTION
**Pull request by cuongdo**
### Description:

Various T3 fixes. Here are the highlights:

* Upgrade component for Google login. Fixes a problem I had.
* Tests for token vesting & transfer code.
* Fix bug where we were transferring an amount in token units, instead of natural units. So, we were transferring far fewer tokens than promised.
* Allow setting of network ID through the env variable NETWORK_ID.
* Very basic server logging.
* Use Sequelize 5.x, like our other packages (e.g. origin-discovery).
* Use underscores for database column names.
### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] ~~Wrap any new text/strings for translation~~
- [ ] ~~Map any new environment variables with a default value in the Webpack config~~
- [ ] ~~Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/origin-docs)~~

### CODE REVIEW

**Comment by cuongdo** [comment]
@tomlinton I'm puzzled by the error I'm getting in the [origin-token-transfer-server tests](https://travis-ci.org/OriginProtocol/origin/jobs/465148980):

```
Error: Cannot find module 'origin-contracts/build/contracts/OriginToken.json'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:580:15)
    at Function.Module._load (internal/modules/cjs/loader.js:506:25)
    at Module.require (internal/modules/cjs/loader.js:636:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/home/travis/build/OriginProtocol/origin/origin-faucet/lib/token.js:3:23)
    at Module._compile (internal/modules/cjs/loader.js:688:30)
    at Module._compile (/home/travis/build/OriginProtocol/origin/node_modules/pirates/lib/index.js:83:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:699:10)
    at Object.newLoader [as .js] (/home/travis/build/OriginProtocol/origin/node_modules/pirates/lib/index.js:88:7)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
    at Module.require (internal/modules/cjs/loader.js:636:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.require (/home/travis/build/OriginProtocol/origin/origin-token-transfer/server/test/lib_transfer.test.js:8:15)
    at Module._compile (internal/modules/cjs/loader.js:688:30)
    at Module._compile (/home/travis/build/OriginProtocol/origin/node_modules/pirates/lib/index.js:83:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:699:10)
    at Object.newLoader [as .js] (/home/travis/build/OriginProtocol/origin/node_modules/pirates/lib/index.js:88:7)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
    at Module.require (internal/modules/cjs/loader.js:636:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at /home/travis/build/OriginProtocol/origin/node_modules/mocha/lib/mocha.js:250:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/home/travis/build/OriginProtocol/origin/node_modules/mocha/lib/mocha.js:247:14)
    at Mocha.run (/home/travis/build/OriginProtocol/origin/node_modules/mocha/lib/mocha.js:576:10)
    at Object.<anonymous> (/home/travis/build/OriginProtocol/origin/node_modules/mocha/bin/_mocha:637:18)
    at Module._compile (internal/modules/cjs/loader.js:688:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:699:10)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:741:12)
    at startup (internal/bootstrap/node.js:285:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:739:3)
```

I've already added bootstrapping for `origin-contracts` to `travis.yml`. Any ideas?

File: origin-token-transfer/server/migrations/20181018102506-create-grant.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 module.exports = {
   up: async (queryInterface, Sequelize) => {
-    await queryInterface.createTable('Grants', {
+    await queryInterface.createTable('Grant', {
```
Comment (franckc) [inline origin-token-transfer/server/migrations/20181018102506-create-grant.js]: super nit: in the rest of the stack, we have use all lower case for table name. e.g. "grants" vs "Grants". Same comment for the events table below.

**Comment by tomlinton** [comment]
@cuongdo that failure is `origin-faucet` attempting to import that contract. Since t3 is importing `origin-faucet` it needs to be bootstrapped as well otherwise lerna won't create the link between `origin-contracts` and `origin-faucet` (even though `origin-contracts` gets built by the bootstrap command). I added it in there.

It doesn't look like `origin-contracts` is a direct dependency here so we can probably remove it from the `package.json`? It'll still need to be part of the bootstrap command though so the contracts get built.

PR looks great!

**Comment by cuongdo** [comment]
Thanks @franckc and @tomlinton for your reviews and help! I'm testing a workaround for a test that runs longer than our default timeout, then I'll merge.

File: origin-faucet/lib/config.js [hunk 2]
```diff
@@ -89,11 +89,15 @@ function createProviders(networkIds) {
     }
     // Private key takes precedence
     if (privateKey) {
-      console.log(`Network=${networkId} URL=${providerUrl} Using private key`)
+      if (process.env.NODE_ENV !== 'test') {
```
[the remark this answers is not in the document]
Comment (cuongdo) [reply origin-faucet/lib/config.js]: Heads up: I'm working on this as part of making the token contract migration tool. The tool will need access to `origin-faucet/lib`, so it makes sense to move that into a new package.

[the remark this answers is not in the document]
Comment (franckc) [reply origin-faucet/lib/config.js]: Yay :)

**Pull request merged**